### PR TITLE
Ghc7.10 - fixes #15

### LINKED ISSUE #15: GHC 7.10 support
I've tried to build **ghci-ng** with GHC version 7.10.1 (modified dependency `ghc == 7.10.1`) and failed:

```
Preprocessing executable 'ghci-ng' for ghci-ng-0.0.0...
[1 of 8] Compiling GhciTypes        ( ghc/GhciTypes.hs, dist/build/ghci-ng/ghci-ng-tmp/GhciTypes.o )
[2 of 8] Compiling GhciMonad        ( ghc/GhciMonad.hs, dist/build/ghci-ng/ghci-ng-tmp/GhciMonad.o )

ghc/GhciMonad.hs:54:1: Warning:
    The import of ‘Control.Applicative’ is redundant
      except perhaps to import instances from ‘Control.Applicative’
    To import instances alone, use: import Control.Applicative()
[3 of 8] Compiling GhciTags         ( ghc/GhciTags.hs, dist/build/ghci-ng/ghci-ng-tmp/GhciTags.o )
[4 of 8] Compiling GhciInfo         ( ghc/GhciInfo.hs, dist/build/ghci-ng/ghci-ng-tmp/GhciInfo.o )

ghc/GhciInfo.hs:185:67:
    Not in scope: type constructor or class ‘PostTcType’
    Perhaps you meant one of these:
      ‘PostTcExpr’ (imported from GHC), ‘PostTcTable’ (imported from GHC)
```

My knowledge do not let me go any further, and I don't know is it hard to make build successful.

Can you give any explanations or clarifications?

### PR DESCRIPTION
**Pull request by schell**
This compiles but I'm not sure if it's "optimal".

### CODE REVIEW

File: ghc/Main.hs [hunk 1]
```diff
@@ -514,7 +523,12 @@ parseModeFlags args = do
              Nothing     -> doMakeMode
              Just (m, _) -> m
       errs = errs1 ++ map (mkGeneralLocated "on the commandline") errs2
-  when (not (null errs)) $ throwGhcException $ errorsToGhcException errs
+  when (not (null errs)) $ throwGhcException
+#if __GLASGOW_HASKELL__ < 709
+                             $ errorsToGhcException errs
+#else
+                             $ errorsToGhcException $ map (\(L sp e) -> (show sp, e)) errs
```
Comment (schell) [inline ghc/Main.hs]: @chrisdone This is questionable - the type `errorsToGhcException` accepts changed from [Located String] to [(String, String)] and the new source calls the fst "Location" and the snd "Error". I'm not sure what a SrcSpan is, though it seems like some source text which may be a location?
https://downloads.haskell.org/~ghc/latest/docs/html/libraries/ghc-7.10.1/src/CmdLineParser.html#errorsToGhcException

File: ghc/Main.hs [hunk 2]
```diff
@@ -522,59 +536,68 @@ type ModeM = CmdLineP (Maybe (Mode, String), [String], [Located String])
...
+               , Flag "E"            (PassFlag (setMode (stopBeforeMode anyHsc)))
+               , Flag "C"            (PassFlag (setMode (stopBeforeMode HCc)))
+               , Flag "S"            (PassFlag (setMode (stopBeforeMode (as False))))
+               , Flag "-make"        (PassFlag (setMode doMakeMode))
+               , Flag "-interactive" (PassFlag (setMode doInteractiveMode))
+               , Flag "-abi-hash"    (PassFlag (setMode doAbiHashMode))
+               , Flag "e"            (SepArg   (\s -> setMode (doEvalMode s) "-e"))
+               ]
+#if __GLASGOW_HASKELL__ >= 709
+        ghcModes = cycle [AllModes]
```
Comment (schell) [inline ghc/Main.hs]: @chrisdone - I just guessed the `AllModes` flag.
https://downloads.haskell.org/~ghc/latest/docs/html/libraries/ghc-7.10.1/CmdLineParser.html#v:AllModes

**Comment by schell** [comment]
I've been using this for about a week with no problems so far. This will fix #15.

**Comment by sinelaw** [comment]
+1 works for me

**Pull request merged**